blocks/image-link-slider/image-link-slider.js

```diff
@@ -0,0 +1,101 @@
+                    </div>
+                    <div class="carousel-slide-content">
+                        <h3><a href="${item.finLink}">${item.finTitle}</a></h3>
+                        <div class="carousel-slide-content-footer copyright">
+                        <div>
+                            <div>${item.finPoweredBy}</div>
+                            <div>${item.finPoweredOn}</div>
+                        </div>
+                        <div class="socialshare">
+                            <a class="social-share">
+                                <img src="/icons/gray-share-icon.svg" alt="Social Share" >
+                            </a>
+                        </div>
+                    </div>
+                    </div>
+                `;
+    carouselItems.append(slide);
+  });
+}
+
+async function loadCarousel(block, carouselItems) {
+  const carouselBlock = buildBlock('carousel', '');
+  carouselBlock.innerHTML = '';
+  block.classList.forEach((className) => {
+    carouselBlock.classList.add(className);
+  });
+
+  Array.from(carouselItems.children).forEach((carouselItemElement) => {
+    const divElement = document.createElement('div');
+    Array.from(carouselItemElement.children).forEach((child) => {
+      divElement.appendChild(child.cloneNode(true));
+    });
+    carouselBlock.appendChild(divElement);
+  });
+
+  const carouselBlockParent = document.createElement('div');
+  carouselBlockParent.classList.add('carousel-wrapper');
+  carouselBlockParent.appendChild(carouselBlock);
+  block.parentElement.insertAdjacentElement('afterend', carouselBlockParent);
+  decorateBlock(carouselBlock);
+
+  carouselBlock.querySelectorAll('.social-share').forEach((anchor) => anchor.addEventListener('click', () => handleSocialShareClick(anchor)));
+  return loadBlock(carouselBlock);
+}
+
+function handleTitleConfig(titleElement, container) {
+  const titleText = titleElement.textContent.trim();
+  const title = document.createElement('h2');
+  title.textContent = titleText;
+
+  const titleWrapper = document.createElement('div');
+  titleWrapper.classList.add('title-wrapper');
+  titleWrapper.appendChild(title);
+
+  container.insertBefore(titleWrapper, container.firstChild);
+}
+
+export default async function decorate(block) {
+  block.style.display = 'none';
+  const container = block.closest('.section .section-container');
+  container.style.display = 'none';
+  const configElementsArray = Array.from(block.children);
+
+  const promises = configElementsArray.map(async (configElement) => {
+    const configNameElement = configElement.querySelector('div');
+    const configName = configNameElement.textContent.trim().toLowerCase();
+    if (configName === 'api') {
+      const carouselItems = document.createElement('div');
+      carouselItems.classList.add('carousel-items');
+      const apiName = configNameElement.nextElementSibling.textContent.trim();
+      const data = await callAPI(apiName);
```

Comment:
Do we need to `await` this api call ? Asking this as this is going to make the block decoration as slow as the api call. Instead, can we just make the api call without awaiting and fill up the block with the api response after it arrives and till then, show a loading spinner or just blank space.
Also copying @anagarwa as we had similar discussion for another block so maybe we can follow a common pattern here.

Reply:
Yes right, it was in previous commit. but got removed by mistake. Thanks for catching it up.